hypervisor/include/dm/vpci.h

```diff
@@ -184,4 +184,28 @@ int32_t vpci_assign_pcidev(struct acrn_vm *tgt_vm, struct acrn_assign_pcidev *pc
 int32_t vpci_deassign_pcidev(struct acrn_vm *tgt_vm, struct acrn_assign_pcidev *pcidev);
 struct pci_vdev *vpci_init_vdev(struct acrn_vpci *vpci, struct acrn_vm_pci_dev_config *dev_config, struct pci_vdev *parent_pf_vdev);
 
+static inline bool is_pci_io_bar(struct pci_vbar *vbar)
+{
+        return ((vbar->bar_type.io_space.indicator == 1U) && (!vbar->is_mem64hi));
+}
+
+static inline bool is_pci_mem_bar(struct pci_vbar *vbar)
+{
+        return ((vbar->is_mem64hi) || ((vbar->bar_type.mem_space.indicator == 0U)));
+}
+
+static inline bool is_pci_reserved_bar(struct pci_vbar *vbar)
```

Comment:
is_pci_reserved_mem_bar is more precise.

Reply:
Because do not need to init vbar for reserved ptdev bar type, just keep that vbar base and size are all 0. Then for next map/unmap operation ignores these reserved bars. is_pci_reserved_bar() should return true if 1. IO bar with bit[1] !=0; 2.Memory bar bit[2:1] is reserved type.

is_pci_reserved_bar() just deal with memory type, it is a bug. Will add IO bar conditions.
+ ||  ((vbar->bar_type.io_space.indicator == 1U) && (vbar->bar_type.io_space.reserved != 0U))